### PR TITLE
Downscale base connector tests

### PR DESCRIPTION
**Pull request by SemionPar**
## Description
Use smaller tables or smaller data sets in general, where appropriate. This should speed up tests for some connectors considerably.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
Could you confirm CI failures?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 1]
```diff
@@ -472,26 +472,26 @@ public void testShowCreateTable()
     public void testView()
     {
         if (!hasBehavior(SUPPORTS_CREATE_VIEW)) {
-            assertQueryFails("CREATE VIEW nation_v AS SELECT * FROM nation", "This connector does not support creating views");
```
Comment (raunaqmorarka) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:475]: `nation` table is quite small, 25 rows, do we need to change this ?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 2]
```diff
@@ -500,19 +500,19 @@ public void testView()
     public void testMaterializedView()
     {
         if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
-            assertQueryFails("CREATE MATERIALIZED VIEW nation_mv AS SELECT * FROM nation", "This connector does not support creating materialized views");
+            assertQueryFails("CREATE MATERIALIZED VIEW region_mv AS SELECT * FROM region", "This connector does not support creating materialized views");
```
Comment (raunaqmorarka) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:503]: `nation` table is quite small, 25 rows, do we need to change this ?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java (hunk 1 above)
Comment (SemionPar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:475]: `testView` and `testView` take 1m and 2m respectively for a certain connector, but again these are not the slowest ones by far, so we can roll these back if you think it is important to have a (slightly) bigger table here.

Comment (raunaqmorarka) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:475]: The change is okay, we don't care about size of table here. I'm just surprised that a difference of 5 vs 25 rows matters.

**Comment by anusudarsan** [comment]
@SemionPar can we mark this ready for review?

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java (hunk 1 above)
Comment (SemionPar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:475]: Totally agree, but for some source systems writes are extremely slow (especially in CI situation where we use smaller instances). As it is not critical, I will roll back these 2.

**Review by raunaqmorarka — APPROVED** [review]

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java [hunk 3]
```diff
@@ -75,38 +75,38 @@ public void testJoinPushdownAutomatic()
...
 
             // with default configuration such join should not be pushed down;
-            // allowed join_to_tables ratio is 1.25 hence join size need to be less than (444_000 + 74_000) * 1.25 == 647_500
+            // allowed join_to_tables ratio is 1.25 hence join size need to be less than (44_400 + 7_400) * 1.25 == 64_750
             assertThat(query(session, format(simpleJoinQuery, left.getName(), right.getName())))
                     .isNotFullyPushedDown(joinOverTableScans());
 
-            // relax allowed ratio to 2.0; base line is 834_000 / (444_000 + 74_000) == 1.61 but we add some margin to cover possible mistakes in NDV calculations.
-            assertThat(query(maxJoinToTablesRatio(session, 2.0), format(simpleJoinQuery, left.getName(), right.getName())))
+            // relax allowed ratio to 3.0; base line is 111_200 / (44_400 + 7_400) == 2.14 but we add some margin to cover possible mistakes in NDV calculations.
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java:93]: where does 111_200 come from? the estimated join size was 83_400 AFAIK.

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java [hunk 4]
```diff
@@ -75,38 +75,38 @@ public void testJoinPushdownAutomatic()
...
                     .build();
             assertThat(query(onlySmallTablesAllowed, format(simpleJoinQuery, left.getName(), right.getName()))).isNotFullyPushedDown(joinOverTableScans());
 
             // a query which constraints size of join output; only join key is in output
             String smallJoinOutputQuery = format("SELECT l.key FROM %s l JOIN %s r ON l.key = r.key", left.getName(), right.getName());
-            // estimated left table size is ~54_000
-            // estimated right table size is ~9_000
-            // estimated join size is ~54_000 (same as left table)
+            // estimated left table size is ~5_400
+            // estimated right table size is ~_900
```
Comment (hashhar) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java:106]:
```suggestion
            // estimated right table size is ~900
```

**Review by hashhar — APPROVED** [review]
LGTM % questions

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java (hunk 3 above)
Comment (SemionPar) [reply plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java:93]: That's what I would expect, too, after consistently decreasing data size 10x, but actually its 111300 (`joinDataSize` when I put the breakpoint in https://github.com/trinodb/trino/blob/095d35adac6697ebdf5ed17b710e4e8a3eb27265/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcJoinPushdownUtil.java#L110), so I changed the numbers to match the reality and relaxed the ration even more.

**Comment by anusudarsan** [comment]
@hashhar who can merge this PR? looks like comments are addressed

File: plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java [hunk 5]
```diff
@@ -75,38 +75,38 @@ public void testJoinPushdownAutomatic()
...
             gatherStats(left.getName());
             gatherStats(right.getName());
 
             String simpleJoinQuery = "SELECT * FROM %s l JOIN %s r ON l.key = r.key";
-            // estimated left table size is ~444_000 bytes
-            // estimated right table size is ~74_000 bytes
-            // estimated join size is ~834_000
+            // estimated left table size is ~44_400 bytes
+            // estimated right table size is ~7_400 bytes
+            // estimated join size is ~83_400
```
Comment (raunaqmorarka) [inline plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseAutomaticJoinPushdownTest.java:86]: Should this be `111_200` ? A few lines later we say that `base line is 111_200`

**Pull request merged**